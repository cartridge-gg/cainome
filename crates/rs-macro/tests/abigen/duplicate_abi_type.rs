#![no_main]
use cainome_rs_macro::abigen;

abigen!(
    MyContract,
    r#"[
        {
            "type": "struct",
            "name": "core::integer::u256",
            "members": [
              {
                "name": "low",
                "type": "core::integer::u128"
              },
              {
                "name": "high",
                "type": "core::integer::u128"
              }
            ]
        }
    ]"#,
    type_aliases {
        core::integer::u256 as MyStruct1;
        core::integer::u256 as MyStruct2;
    }
);
