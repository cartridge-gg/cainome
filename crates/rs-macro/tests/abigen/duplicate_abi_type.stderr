error: core::integer::u256 duplicate abi type
  --> tests/abigen/duplicate_abi_type.rs:24:9
   |
24 |         core::integer::u256 as MyStruct2;
   |         ^^^^
